app/gui/view/component-browser/component-list-panel/breadcrumbs/src/entry.rs

```diff
@@ -216,15 +216,19 @@ impl EntryData {
         self.state.get() == State::Text
     }
 
-    /// Return the width of the entry if it is known. If the entry displays text, the width needs to
-    /// be calculated separately.
-    fn fixed_width(&self) -> Option<f32> {
+    fn width(&self, text_offset: f32) -> f32 {
         match self.state.get() {
-            State::Text => None,
-            State::Separator => Some(separator::ICON_WIDTH),
-            State::Ellipsis => Some(ellipsis::ICON_WIDTH),
+            State::Text => self.text_width(self.text.width.value(), text_offset),
+            State::Separator => separator::ICON_WIDTH,
+            State::Ellipsis => ellipsis::ICON_WIDTH,
         }
     }
+
+    /// Width of the breadcrumb column filled with text of width [`text_width`] and with offset
+    /// [`text_offset`].
+    fn text_width(&self, text_width: f32, text_offset: f32) -> f32 {
+        text_width + text_offset * 2.0
+    }
```

Comment:
Why do we double the offset? Maybe it should be called "margin"?

Reply:
We're pretty inconsistent in using the terms "margin", "offset" and "padding" throughout the codebase, but you're right. I changed it to `padding` (as this name was initially used in the stylesheet, and I think I copied it from another place in the code).